### PR TITLE
feat: deno ci

### PR DESCRIPTION
**Pull request by marco-ippolito**
Resolves: https://github.com/LyraSearch/lyra/issues/187
I'm creating this pull request to support a ci for deno
It seems to be working although setup-deno action does not support natively caching like setup-node.
There is a issue on this topic on setup-deno: https://github.com/denoland/setup-deno/issues/31
Here a successful run on the fork: https://github.com/marco-ippolito/lyra/actions/runs/3663984022
I've also renamed the github actions for more clarity and corrected a typo.
Also I'm not sure this is the correct approach...

### CODE REVIEW

File: .github/workflows/deno-ci.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Tests
```
Comment (micheleriva) [inline .github/workflows/deno-ci.yml:1]: Could you plese rename to `Deno Tests`?

**Comment by marco-ippolito** [comment]
I've added a broken package to make sure pnpm behaviour was correct https://github.com/LyraSearch/lyra/actions/runs/3693217103/jobs/6252948194

**Comment by micheleriva** [comment]
@marco-ippolito looks too good to be true 🙂 I guess we can merge it! Could you revert your last commit?

**Comment by marco-ippolito** [comment]
its using node :( 

```
> c8 -c tests/config/c8-local.json tap --rcfile=tests/config/tap.yml tests/*.test.ts


    process arg /usr/local/bin/node
tests/deno.test.ts
```

**Pull request closed without merging**